internal/stanza/emitter.go

```diff
@@ -17,46 +17,181 @@ package stanza // import "github.com/open-telemetry/opentelemetry-collector-cont
 import (
 	"context"
 	"sync"
+	"time"
 
 	"github.com/open-telemetry/opentelemetry-log-collection/entry"
+	"github.com/open-telemetry/opentelemetry-log-collection/operator"
 	"github.com/open-telemetry/opentelemetry-log-collection/operator/helper"
 	"go.uber.org/zap"
 )
 
 // LogEmitter is a stanza operator that emits log entries to a channel
 type LogEmitter struct {
 	helper.OutputOperator
-	logChan  chan *entry.Entry
-	stopOnce sync.Once
+	logChan       chan []*entry.Entry
+	stopOnce      sync.Once
+	cancel        context.CancelFunc
+	batchMux      sync.Mutex
+	batch         []*entry.Entry
+	wg            sync.WaitGroup
+	maxBatchSize  uint
+	flushInterval time.Duration
+}
+
+type LogEmitterOption interface {
+	Apply(*LogEmitter)
+}
+
+type logEmitterOptionFunc func(*LogEmitter)
+
+func (leo logEmitterOptionFunc) Apply(le *LogEmitter) {
```

Comment:
This feels a bit inverted to me. Why not have `func (le *LogEmitter) Apply(leo logEmitterOptionFunc)`?

Reply:
I heavily agree; I copied the pattern used for functional options in the converter, which does basically the same thing here; I wanted to match it for consistency sake, but it is definitely unnecessary.

I refactored it for the emitter, should I also refactor the converter to be in line with the emitter here?

Reply:
> should I also refactor the converter to be in line with the emitter here?

In a follow up PR would be best